dashboard/permissions.py

```diff
@@ -0,0 +1,39 @@
+"""
+Permission classes for the dashboard
+"""
+from django.contrib.auth.models import User
+from django.http import Http404
+from rest_framework.generics import get_object_or_404
+from rest_framework.permissions import BasePermission
+
+from roles.roles import (
+    Instructor,
+    Staff,
+)
+
+
+class CanReadIfStaffOrSelf(BasePermission):
+    """
+    Only staff on a program the learner is enrolled in can
+    see their dashboard. Learners can view their own dashboard.
+    """
+
+    def has_permission(self, request, view):
+        if request.user.is_anonymous():
+            raise Http404
+
+        user = get_object_or_404(User, username=view.kwargs['username'])
```

Comment:
Yes, the only username we expose to the outside user is their edX username. Usually this will be the same as `User.username` but if there's a conflict or an error when creating the social auth object it may create a separate user object with some hex digits appended to the username, ie `george6fb83a` for an edX username of `george`. You can look it up by joining the social auth table like this: `User.objects.filter(social_auth__uid='a_username_here')`